agent/fw_yii.c

```diff
@@ -86,21 +86,134 @@ NR_PHP_WRAPPER(nr_yii_runWithParams_wrapper) {
 NR_PHP_WRAPPER_END
 
 /*
- * Enable Yii instrumentation.
+ * Enable Yii1 instrumentation.
  */
-void nr_yii_enable(TSRMLS_D) {
+void nr_yii1_enable(TSRMLS_D) {
 #if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
     && !defined OVERWRITE_ZEND_EXECUTE_DATA
   nr_php_wrap_user_function_before_after_clean(
-      NR_PSTR("CAction::runWithParams"), nr_yii_runWithParams_wrapper, NULL,
+      NR_PSTR("CAction::runWithParams"), nr_yii1_runWithParams_wrapper, NULL,
       NULL);
   nr_php_wrap_user_function_before_after_clean(
-      NR_PSTR("CInlineAction::runWithParams"), nr_yii_runWithParams_wrapper,
+      NR_PSTR("CInlineAction::runWithParams"), nr_yii1_runWithParams_wrapper,
       NULL, NULL);
 #else
   nr_php_wrap_user_function(NR_PSTR("CAction::runWithParams"),
-                            nr_yii_runWithParams_wrapper TSRMLS_CC);
+                            nr_yii1_runWithParams_wrapper TSRMLS_CC);
   nr_php_wrap_user_function(NR_PSTR("CInlineAction::runWithParams"),
-                            nr_yii_runWithParams_wrapper TSRMLS_CC);
+                            nr_yii1_runWithParams_wrapper TSRMLS_CC);
+#endif
+}
+
+/*
+ * Yii2: Set the web transaction name from the unique action ID.
+ */
+NR_PHP_WRAPPER(nr_yii2_runWithParams_wrapper) {
+  zval* this_var = NULL;
+  zval* unique_idz = NULL;
+  const char* unique_id = NULL;
+  nr_string_len_t unique_id_length;
+  char* transaction_name = NULL;
+
+  (void)wraprec;
+  NR_UNUSED_SPECIALFN;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_YII2);
+
+  this_var = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (NULL == this_var) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "Yii2: improper this");
+    goto end;
+  }
+
+  unique_idz = nr_php_call(this_var, "getUniqueId");
```

Comment:
do we need to call `nr_php_zval_free` on this value?

Reply:
I followed the same logic as above, where `classz = nr_php_call(this_var, "getController");` is not freed. Not sure if needed tho....

Reply:
None of the tests catch it but it does look like a memory leak. The zval returned by `nr_php_call` is allocated here: https://github.com/newrelic/newrelic-php-agent/blob/51c11405c1efb530fe60ea178398f9450198ec74/agent/php_call.c#L51 Doc strings clearly say it should be destroyed: https://github.com/newrelic/newrelic-php-agent/blob/51c11405c1efb530fe60ea178398f9450198ec74/agent/php_call.h#L20-L21 Patched with dc5fa2c89e28140e8a4a1acd802690778f137b53.